### PR TITLE
feat(chat): implement isolated model view page (Issue #1154)

### PR DESCRIPTION
**Pull request by i-serebryakov07**
**Description:**

Adding new page /models/:modelId to provide a possibility to open new sessions for selected models with just chat functionality.

Issues:

- Issue #1154

**UI changes**

image

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1152.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1152.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1152.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/pages/models/[modelId].tsx [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+            className="button button-secondary"
+            disabled={authStatus === 'loading'}
+          >
+            {t('Login')}
+          </button>
+        </div>
+      ) : (
+        <>
+          {modelId && modelIsLoaded && !activeModel ? (
+            <div className="pt-2">
```
Comment (IlyaBondar) [inline apps/chat/src/pages/models/[modelId].tsx]:
```suggestion
            <div className="h-screen pt-2">
```
->
image

File: apps/chat/src/utils/server/api.ts [hunk 2]
```diff
@@ -101,7 +101,8 @@ export class ApiUtils {
     constructPath(...path.split('/').map((part) => decodeURIComponent(part)));
 
   static request(url: string, options?: RequestInit) {
-    return fromFetch(url, {
+    const absoluteUrl = url.startsWith('/') ? url : `/${url}`;
+    return fromFetch(absoluteUrl, {
```
Comment (IlyaBondar) [inline apps/chat/src/utils/server/api.ts]:
```
function isUrlAbsolute(url) { 
    return (url.indexOf('://') > 0 || url.indexOf('//') === 0);
}
```
and
```suggestion
    const correctUrl = !isUrlAbsolute(url) && !url.startsWith('/') ? `/${url}` : url;
    return fromFetch(correctUrl, {
```

File: apps/chat/src/store/conversations/conversations.epics.ts [hunk 3]
```diff
@@ -2571,6 +2623,32 @@ const getChartAttachmentEpic: AppEpic = (action$) =>
...
+        state$.value,
+      );
+      const isolatedModelId = SettingsSelectors.selectIsolatedModelId(
+        state$.value,
+      );
+      if (isIsolatedView && isolatedModelId) {
+        return of(
+          ConversationsActions.deleteConversations({
+            conversationIds: [
+              `conversations/${BucketService.getBucket()}/${isolatedModelId}__isolated_${isolatedModelId}`,
```
Comment (IlyaBondar) [inline apps/chat/src/store/conversations/conversations.epics.ts]:
```suggestion
             constructPath(getConversationRootId(), [
                 encodeModelId(isolatedModelId),
                 `isolated_${isolatedModelId}`,
             ].join(pathKeySeparator)),
```

**Comment by nepalevov** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1152.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/pages/models/[modelId].tsx (hunk 1 above)
Comment (i-serebryakov07) [reply apps/chat/src/pages/models/[modelId].tsx]: Done

File: apps/chat/src/store/conversations/conversations.epics.ts (hunk 3 above)
Comment (i-serebryakov07) [reply apps/chat/src/store/conversations/conversations.epics.ts]: Done

File: apps/chat/src/utils/server/api.ts (hunk 2 above)
Comment (i-serebryakov07) [reply apps/chat/src/utils/server/api.ts]: Done

**Comment by IlyaBondar** [comment]
/deploy-review 
>Environment URL: https://chat-ai-dial-chat-pr-1152.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/store/conversations/conversations.epics.ts [hunk 4]
```diff
@@ -2571,6 +2624,38 @@ const getChartAttachmentEpic: AppEpic = (action$) =>
...
+                  encodeModelId(isolatedModelId),
+                  `isolated_${isolatedModelId}`,
+                ].join(pathKeySeparator),
+              ),
+            ],
+            suppressErrorMessage: true,
+          }),
+        );
+      }
+      return concat();
```
Comment (IlyaBondar) [inline apps/chat/src/store/conversations/conversations.epics.ts]:
```suggestion
      return EMPTY;
```

Comment (i-serebryakov07) [reply apps/chat/src/store/conversations/conversations.epics.ts]: Done, thanks!

**Pull request merged**